lib/ansible/modules/cloud/amazon/aws_s3_cors.py

```diff
@@ -140,11 +140,20 @@ def create_or_update_bucket_cors(connection, module):
 def destroy_bucket_cors(connection, module):
 
     name = module.params.get("name")
+    exists = False
     changed = False
 
     try:
-        cors = connection.delete_bucket_cors(Bucket=name)
-        changed = True
+        connection.get_bucket_cors(Bucket=name)
+        exists = True
+    except ClientError:   # NoSuchCORSConfiguration
```

Comment:
If this has a specific error code it would be better to handle it alone. Otherwise, you may mask legitimate ClientErrors.
```
except ClientError as e:
    # please test this because AWS is inconsistent; I think the error code will either be `404` or `NoSuchCORSConfiguration`
    if e.response['Error']['Code'] == 'NoSuchCORSConfiguration': 
        exists = False
    else:
        # call fail_json because something wrong happened
    
```

Reply:
That looks right @s-hertel:

```
{"Error": {"Code": "NoSuchCORSConfiguration", "Message": "The CORS configuration does not exist", "BucketName": "redacted"}, "ResponseMetadata": {"RequestId": "26E272CC9605E33B", "HostId": "BTIJykHigZ51NeHowsPXd7ad1O/pjgDIqot1fQNBwEvVCSP/EqqUTVzvSgbVuakN+Bt+0pbq2V0=", "HTTPStatusCode": 404, "HTTPHeaders": {"x-amz-request-id": "26E272CC9605E33B", "x-amz-id-2": "BTIJykHigZ51NeHowsPXd7ad1O/pjgDIqot1fQNBwEvVCSP/EqqUTVzvSgbVuakN+Bt+0pbq2V0=", "content-type": "application/xml", "transfer-encoding": "chunked", "date": "Mon, 22 Jul 2019 23:37:26 GMT", "server": "AmazonS3"}, "RetryAttempts": 0}}
```

Some clients have errors that are more usable (e.g. `client.exceptions.NoSuchCORSConfigurationError`) but apparently not this error in this client:

```
<botocore.errorfactory.S3Exceptions object at 0x10e0b1b70> object has no attribute 'NoSuchCORSConfigurationError'. Valid exceptions are: BucketAlreadyExists, BucketAlreadyOwnedByYou, NoSuchBucket, NoSuchKey, NoSuchUpload, ObjectAlreadyInActiveTierError, ObjectNotInActiveTierError
```